src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelEMAlgorithm.java

```diff
@@ -0,0 +1,497 @@
+                }
+
+                prevMStepLikelihood = latestMStepLikelihood;
+                latestMStepLikelihood = iterInfo.logLikelihood;
+
+                /* if the likelihood has increased and the increment is small, start updating copy number posteriors */
+                if (params.copyRatioUpdateEnabled() &&
+                        !updateCopyRatioPosteriors &&
+                        (latestMStepLikelihood - prevMStepLikelihood) > 0 &&
+                        (latestMStepLikelihood - prevMStepLikelihood) < params.getLogLikelihoodTolThresholdCRCalling()) {
+                    updateCopyRatioPosteriors = true;
+                    logger.info("Partial convergence achieved; starting to update copy ratio posteriors (if enabled)" +
+                            " and sample-specific unexplained variance (if enabled) in the next iteration");
+                    if (params.adaptivePsiSolverModeSwitchingEnabled()) {
+                        params.setPsiPsiolverType(CoverageModelEMParams.PsiUpdateMode.PSI_TARGET_RESOLVED);
+                    }
+                }
+            }
+
+            /* check convergence in log likelihood change */
+            if (FastMath.abs(latestMStepLikelihood - prevMStepLikelihood) < params.getLogLikelihoodTolerance()) {
+                /* make sure that we have either already called copy ratio posteriors, or we are not required to */
+                if (!params.copyRatioUpdateEnabled() || updateCopyRatioPosteriors) {
+                    status = EMAlgorithmStatus.SUCCESS_LIKELIHOOD_TOL;
+                    break;
+                }
+            } else if (iterInfo.iter == params.getMaxEMIterations() - 2) {
+                performMStep = false; /* so that we end with an E-step */
+            }
+
+            iterInfo.increaseIterationCount();
+
+            finalizeIteration();
+
+            /* checkpointing */
+            if (params.isRunCheckpointingEnabled() && iterInfo.iter % params.getRunCheckpointingInterval() == 0) {
+                final String modelOutputAbsolutePath = new File(params.getRunCheckpointingPath(),
+                        String.format("%s_iter_%d", CoverageModelGlobalConstants.MODEL_CHECKPOINT_PATH_PREFIX, iterInfo.iter)).getAbsolutePath();
+                final String posteriorOutputAbsolutePath = new File(params.getRunCheckpointingPath(),
+                        String.format("%s_iter_%d", CoverageModelGlobalConstants.POSTERIOR_CHECKPOINT_PATH_PREFIX, iterInfo.iter)).getAbsolutePath();
+                saveModel(modelOutputAbsolutePath);
+                savePosteriors(posteriorOutputAbsolutePath, PosteriorVerbosityLevel.BASIC);
+            }
+        }
+
+        if (iterInfo.iter == params.getMaxEMIterations()) {
+            status = EMAlgorithmStatus.FAILURE_MAX_ITERS_REACHED;
+        }
+
+        performPostEMOperations();
+
+        return status;
+    }
+
+    /**
+     *
+     */
+    public EMAlgorithmStatus runExpectation() {
```

Comment:
It was originally like that but `runExpectationMaximization` has too many knobs to disable to make it look like `runExpectation`. The code is way more readable this way.